### PR TITLE
[New feature] Mouseover event for VDatePicker #4884

### LINKED ISSUE #4884: [Feature Request] Popup events on mouseover for VDatePicker
### What problem does this feature solve?
Users should be able to get a top level description of the events for the mouseover-ed date.

### What is your proposed solution?
Supply additional event @mouseover with the hovered date so the parent could display the details for the date.

Component: VDatePicker

### PR DESCRIPTION
**Pull request by Kailaash-Balachandran**
## Description
Added two additional events @date-mouse-enter and @date-mouse-leave that returns the date value on hover.

`<v-date-picker @date-mouse-enter="mouseEnter($event)" @date-mouse-leave="mouseLeave($event)"`

## Motivation and Context
Idea is to display the list of events for the hovered date in a small popup. On date click, user be will shown the detailed view. Currently, v-date-picker only supports @input. Thus the events introduced in this MR, @date-mouse-enter and @date-mouse-leave, could come in handy for such use cases.

fixes #4884 

## How Has This Been Tested?
Added new unit test cases

## Markup:

<details>

```vue
<template>
  <v-app>
    <v-date-picker 
      @date-mouse-enter="mouseEnter($event)" 
      @date-mouse-leave="mouseLeave($event)">
    </v-date-picker>
  </v-app>
</template>

<script>
  export default {
    methods: {
      mouseEnter(dateValue) {
        //Event handler
      },
      mouseLeave (dateValue) {
        //Event handler
      }
    }
  }
</script>

```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

File: src/components/VDatePicker/VDatePicker.js [hunk 1]
```diff
@@ -288,6 +287,22 @@ export default {
...
+      this.inputMonth = parseInt(value.split('-')[1], 10) - 1
+      this.inputDay = parseInt(value.split('-')[2], 10)
+      const output = this.multiple
+        ? (
+          this.value.indexOf(value) === -1
+            ? this.value.concat([value])
+            : this.value.filter(x => x !== value)
+        )
+        : value
+      this.$emit('mouseover', output)
```
Comment (jacekkarczmarczyk) [inline src/components/VDatePicker/VDatePicker.js]: Why not just emit value? Also use the same event name (mouseenter)

File: src/components/VDatePicker/VDatePicker.js [hunk 2]
```diff
@@ -288,6 +287,22 @@ export default {
...
+    },
+    mouseLeave () {
+      this.$emit('mouseover', null)
```
Comment (jacekkarczmarczyk) [inline src/components/VDatePicker/VDatePicker.js]: Mouseover should be mouseleave and should also send a value

File: src/components/VDatePicker/mixins/date-picker-table.js [hunk 3]
```diff
@@ -85,7 +85,9 @@ export default {
           innerHTML: `<div class="v-btn__content">${this.formatter(value)}</div>`
         },
         on: (this.disabled || !isAllowed) ? {} : {
-          click: () => this.$emit('input', value)
+          click: () => this.$emit('input', value),
+          mouseenter: () => this.$emit('mouseenter', value),
+          mouseleave: () => this.$emit('mouseleave')
```
Comment (jacekkarczmarczyk) [inline src/components/VDatePicker/mixins/date-picker-table.js]: Send a value

File: src/components/VDatePicker/VDatePicker.js (hunk 1 above)
Comment (jacekkarczmarczyk) [reply src/components/VDatePicker/VDatePicker.js]: Oh i see what you did here, so probably using dom events names is not the best idea, still I think that there should be 2 events ("*enter" and "*leave"), both sending the value, but I'd change the naming. Maybe `date-mouse-enter` and `date-mouse-leave` or something similar?

Comment (Kailaash-Balachandran) [reply src/components/VDatePicker/VDatePicker.js]: The idea was to keep the implementation simple with just @mouseover dom event that returns the date value or null (while leaving). But yes, I agree there should be two events. Event names date-mouse-enter and date-mouse-leave sounds good to me.

File: src/components/VDatePicker/mixins/date-picker-table.js [hunk 4]
```diff
@@ -85,7 +85,9 @@ export default {
           innerHTML: `<div class="v-btn__content">${this.formatter(value)}</div>`
         },
         on: (this.disabled || !isAllowed) ? {} : {
-          click: () => this.$emit('input', value)
+          click: () => this.$emit('input', value),
+          mouseenter: () => this.$emit('mouseenter', value),
```
Comment (jacekkarczmarczyk) [inline src/components/VDatePicker/mixins/date-picker-table.js]: Please use `mouse-date-enter/leave` names here as well and we'll merge it

Comment (Kailaash-Balachandran) [reply src/components/VDatePicker/mixins/date-picker-table.js]: Done..

**Comment by jacekkarczmarczyk** [comment]
@Kailaash-Balachandran looks good, but i think that events should be fired also for month picker (only when date picker has `type` prop set to "month"). Please add it so it could land in 1.3

**Comment by johnleider** [comment]
Closing due to inactivity

**Comment by Kailaash-Balachandran** [comment]
@jacekkarczmarczyk I would like to restart working on this feature. Is someone else already working on it?

**Comment by jacekkarczmarczyk** [comment]
I'll start working on this after merging (or closing) this one https://github.com/vuetifyjs/vuetify/pull/4906

**Pull request closed without merging**